apps/entity-service/src/entities/file-upload.controller.ts

```diff
@@ -0,0 +1,30 @@
+import { Controller, Post, Param, NotFoundException, Req } from "@nestjs/common";
+import { FileUploadService } from "../trees/file-upload.service";
+import { EntitiesService } from "./entities.service";
+import { EntityModel } from "@terramatch-microservices/database/constants/entities";
+import { SpecificEntityDto } from "./dto/specific-entity.dto";
+import { PolicyService } from "@terramatch-microservices/common/policies/policy.service";
+import { FastifyRequest } from "fastify";
+
+@Controller("entities/v3/file/upload")
+export class FileUploadController {
+  constructor(
+    private readonly fileUploadService: FileUploadService,
+    private readonly entitiesService: EntitiesService,
+    private readonly policyService: PolicyService
+  ) {}
+
+  @Post(":collection/:entity/:uuid")
+  async uploadFile<T extends EntityModel>(
+    @Param("collection") collection: string,
+    @Param() { entity, uuid }: SpecificEntityDto,
```

Comment:
I kind of dislike using the `@Param("name")` construct because it doesn't allow for doing validation (not super important here, but as a pattern I think we should avoid it). Also, I don't think there's any real reason to go out of the way to re-use DTOs from the base entities service. I'd create a param DTO specific to this endpoint and use it here to collect all three. It could extend the `SpecificEntityDto` to avoid some duplication if you like.